### DISCUSSION TITLE
How to "add" a many to many relation to a given collection using graphql

### DISCUSSION THREAD

**Discussion by cpt-n3mo** [directus/directus#11694] [Archive]
Good day,

i have the following example..

a collection: customers, has a many to many relation with a collection payments,
how do i add a new payment to the customer collection via graphql

if i do the following it just replaces the already set payment relations with the new one, and is not adding it.

```
mutation {
  update_customer_item(
    id: 2, data:{
      customer_payments:{
        payments_id:{
          test: "new item"
        }
      }
    }
  ){
    id
  }
}

```
kind regards,

**Comment by joselcvarela** [comment]
Hello,
I believe you need to reference the junction table. 
For example:
```graphql
mutation {
  update_customer_item(
    id: 2, data:{
      customer_payments:{
        payments: {
          payments_id:{
            test: "new item"
          }
        }
      }
    }
  ){
    id
  }
}

```

**Reply by cpt-n3mo** [reply]
Good day!
thx for your responce,

sadly this does not seem to work.
the only way i could get this to work is by doing the following. if i add a a existing id to it like this it will work. but this would need 1 query and a mutation. a query to get the current relations ids, and then a mutation using those ids + the data to be added via a mutation..
this is def not ideal but its a way i could get it to work..

maybe there could be some additions to the api..?
like create, connect, remove, for relations,
so it becomes easier to, add, connect, remove. connect is basically there by the use of adding a existing id, but create and remove are not,..

```
mutation {
  update_customer_item(
    id: 2, data:{
      customer_payments:[{
        payments_id:{
          test: "new item"
        }
      }
      {
        payments_id:{
          id: "1"
        }
      }
      ]
      }
  ){
    id
  }
}

```

kind regards

**Comment by rijkvanzanten** [comment]
Lets keep discussing this on https://github.com/directus/directus/discussions/11773 to not have multiple threads going at the same time 👍🏻
